signer/src/block_observer.rs

```diff
@@ -63,21 +61,20 @@ pub struct BlockObserver<BitcoinClient, StacksClient, EmilyClient, BlockHashStre
     pub network: bitcoin::Network,
 }
 
-impl<BC, SC, EC, BHS, S> BlockObserver<BC, SC, EC, BHS, S>
+impl<SC, EC, BHS, S> BlockObserver<SC, EC, BHS, S>
 where
-    BC: BitcoinInteract + BitcoinClient,
     SC: StacksInteract,
     EC: EmilyInteract,
     S: DbWrite + DbRead + Send + Sync,
     BHS: futures::stream::Stream<Item = bitcoin::BlockHash> + Unpin,
 {
     /// Run the block observer
-    #[tracing::instrument(skip(self))]
-    pub async fn run(mut self) -> Result<(), Error> {
+    #[tracing::instrument(skip(self, ctx))]
+    pub async fn run(mut self, ctx: impl Context) -> Result<(), Error> {
```

Comment:
Ahh, so it looks like this PR moves us to having all (or most) of the clients encapsulated in the context?

Reply:
At least for right now -- I've been a bit on the fence, but otoh these clients will only be used in the signer and are created from the config, so it doesn't _not_ make sense to avoid needing to new these up for every dependency :shrug:

Reply:
> [I]t doesn't _not_ make sense to avoid needing to manually new these up for every dependency 🤷 It also lets us share the current "alive" endpoint so we avoid unnecessary retries.
>
> I figure that if it ends up not feeling right, it's not a big refactor to move them out again.

Yeah, that all makes sense me.